src/app/clusters/actions-server/actions-server.h

```diff
@@ -0,0 +1,341 @@
+
+class Delegate;
+
+struct ActionStructStorage : public Structs::ActionStruct::Type
+{
+    ActionStructStorage() : mActionName(mBuffer, sizeof(mBuffer)){};
+
+    ActionStructStorage(uint16_t action, CharSpan actionName, ActionTypeEnum actionType, uint16_t epListID,
+                        BitMask<CommandBits> commands, ActionStateEnum actionState) :
+        mActionName(mBuffer, sizeof(mBuffer))
+    {
+        Set(action, actionName, actionType, epListID, commands, actionState);
+    }
+
+    ActionStructStorage(const ActionStructStorage & action) : mActionName(mBuffer, sizeof(mBuffer)) { *this = action; }
+
+    ActionStructStorage & operator=(const ActionStructStorage & action)
+    {
+        Set(action.actionID, action.name, action.type, action.endpointListID, action.supportedCommands, action.state);
+        return *this;
+    }
+
+    void Set(uint16_t action, CharSpan actionName, ActionTypeEnum actionType, uint16_t epListID, BitMask<CommandBits> commands,
+             ActionStateEnum actionState)
+    {
+        actionID          = action;
+        type              = actionType;
+        endpointListID    = epListID;
+        supportedCommands = commands;
+        state             = actionState;
+        CopyCharSpanToMutableCharSpanWithTruncation(actionName, mActionName);
+        name = mActionName;
+    }
+
+private:
+    char mBuffer[kActionNameMaxSize];
+    MutableCharSpan mActionName;
+};
+
+struct EndpointListStorage : public Structs::EndpointListStruct::Type
+{
+    EndpointListStorage() : mEpListName(mBuffer, sizeof(mBuffer)){};
+
+    EndpointListStorage(uint16_t epListId, CharSpan epListName, EndpointListTypeEnum epListType,
+                        DataModel::List<const EndpointId> endpointList) :
+        mEpListName(mBuffer, sizeof(mBuffer))
+    {
+        Set(epListId, epListName, epListType, endpointList);
+    }
+
+    EndpointListStorage(const EndpointListStorage & epList) : mEpListName(mBuffer, sizeof(mBuffer)) { *this = epList; }
+
+    EndpointListStorage & operator=(const EndpointListStorage & epList)
+    {
+        Set(epList.endpointListID, epList.name, epList.type, epList.endpoints);
+        return *this;
+    }
+
+    void Set(uint16_t epListId, CharSpan epListName, EndpointListTypeEnum epListType,
+             DataModel::List<const EndpointId> endpointList)
+    {
+        endpointListID = epListId;
+        type           = epListType;
+
+        for (uint8_t index = 0; index < std::min(endpointList.size(), kEndpointListMaxSize); index++)
+        {
+            mEpList.push_back(endpointList[index]);
+        }
+        mEpListSpan = Span(mEpList.data(), mEpList.size());
+        endpoints   = DataModel::List<const EndpointId>(mEpListSpan);
+        CopyCharSpanToMutableCharSpanWithTruncation(epListName, mEpListName);
+        name = mEpListName;
+    }
+
+private:
+    char mBuffer[kEndpointListNameMaxSize];
+    MutableCharSpan mEpListName;
+    std::vector<EndpointId> mEpList;
```

Comment:
OK, so why are we ok with a max-sized static buffer for mBuffer (which should be mNameBuffer anyway), but not for the endpoint list?  At the very least that needs to be documented.

Reply:
Used statically allocated array.